src/operator/nn/mkldnn/mkldnn_deconvolution.cc

```diff
@@ -32,6 +32,12 @@
 namespace mxnet {
 namespace op {
 
+bool SupportMKLDNNConv(const DeconvolutionParam& params, const NDArray &input) {
+  if (params.kernel.ndim() != 2)
```

Comment:
do we need to add check for strides and dilate too ?

Reply:
I think we should have a check in the parameter parser of mxnet conv, so we don't need to check it in the MKLDNN code.

Reply:
If we are checking strides and dilates ndim to be greater than equal to 2, can we fallback to default implementation and return false here when ndim of stride, pad or dilates is less than 2 ?